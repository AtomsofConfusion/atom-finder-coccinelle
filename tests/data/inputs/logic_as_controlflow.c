#include <stdio.h>

int fun1() {
    return 1;
}

int fun2() {
    return 1;
}

int fun3 (int* a) {
  a++;
  return *a;
}

int fun4 (int a) {
  a++;
  return a;
}

int fun5 (int* a, int b) {
  int c;
  c = b || (*a += 2);
  return c;
}

int main() {
    int a = 5;
    int b = 4;
    int c = 3;
    int d = 2;
    int e = 1;

    a || b++ & c;

    a && c | --b;

    a && c + b-- & d;

    a || b;

    a || b && c;

    a || b++ & c;

    a && ++b;

    a || b || c++;

    a && b || c++ < 1;

    a && ++b & c;

    a || b-- || c-- && d++ || --e;

    a || b-- || ++c;

    a || fun1(); // no?

    a || !(b = fun1());

    a && (b += 2) || c++;

    int f = a || b-- || c-- && d++ || --e;

    int g = a || (b = fun1());

    if (a && b) {
        a = 6;
    } else if (!a || --a > b) {
        a = 7;
    }

    (a > b) && fun1() && fun2();

    (a > b) || fun1();

    b || fun3(&a);

    b || fun4(a);

    b && (c + fun5(&a, b)); // all of these

}
