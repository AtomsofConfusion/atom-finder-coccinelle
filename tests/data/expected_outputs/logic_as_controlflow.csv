,,23,,b || (*a += 2),coccinelle missing
assignment-as-value,/home/veering/Projects/atom-finder-coccinelle/tests/data/inputs/logic_as_controlflow.c,34,4,a || b ++ & c,
assignment-as-value,/home/veering/Projects/atom-finder-coccinelle/tests/data/inputs/logic_as_controlflow.c,36,4,a && c | -- b,
assignment-as-value,/home/veering/Projects/atom-finder-coccinelle/tests/data/inputs/logic_as_controlflow.c,38,4,a && c + b -- & d,
assignment-as-value,/home/veering/Projects/atom-finder-coccinelle/tests/data/inputs/logic_as_controlflow.c,44,4,a || b ++ & c,
assignment-as-value,/home/veering/Projects/atom-finder-coccinelle/tests/data/inputs/logic_as_controlflow.c,46,4,a && ++ b,
assignment-as-value,/home/veering/Projects/atom-finder-coccinelle/tests/data/inputs/logic_as_controlflow.c,48,4,a || b || c ++,
assignment-as-value,/home/veering/Projects/atom-finder-coccinelle/tests/data/inputs/logic_as_controlflow.c,50,4,a && b || c ++ < 1,
assignment-as-value,/home/veering/Projects/atom-finder-coccinelle/tests/data/inputs/logic_as_controlflow.c,52,4,a && ++ b & c,
assignment-as-value,/home/veering/Projects/atom-finder-coccinelle/tests/data/inputs/logic_as_controlflow.c,54,16,c -- && d ++,
assignment-as-value,/home/veering/Projects/atom-finder-coccinelle/tests/data/inputs/logic_as_controlflow.c,54,4,a || b -- || c -- && d ++,
assignment-as-value,/home/veering/Projects/atom-finder-coccinelle/tests/data/inputs/logic_as_controlflow.c,56,4,a || b --,
,,58,,a || fun1(),both missing
assignment-as-value,/home/veering/Projects/atom-finder-coccinelle/tests/data/inputs/logic_as_controlflow.c,60,4,a || ! ( b = fun1 ( ) ),
assignment-as-value,/home/veering/Projects/atom-finder-coccinelle/tests/data/inputs/logic_as_controlflow.c,62,4,a && ( b += 2 ) || c ++,
assignment-as-value,/home/veering/Projects/atom-finder-coccinelle/tests/data/inputs/logic_as_controlflow.c,64,24,c -- && d ++,
assignment-as-value,/home/veering/Projects/atom-finder-coccinelle/tests/data/inputs/logic_as_controlflow.c,64,12,a || b -- || c -- && d ++,
assignment-as-value,/home/veering/Projects/atom-finder-coccinelle/tests/data/inputs/logic_as_controlflow.c,66,12,a || ( b = fun1 ( ) ),
assignment-as-value,/home/veering/Projects/atom-finder-coccinelle/tests/data/inputs/logic_as_controlflow.c,70,15,! a || -- a > b,
,,74,,(a > b) && fun1() && fun2(),both missing
,,76,,(a > b) || fun1(),both missing
,,78,,b || fun3(&a),both missing
,,80,,b || fun4(a),both missing
assignment-as-value,/home/veering/Projects/atom-finder-coccinelle/tests/data/inputs/logic_as_controlflow.c,82,4,"b && ( c + fun5 ( & a , b ) )",
